src/tree/context-dep.cc

```diff
@@ -179,6 +179,50 @@ void ContextDependency::Read (std::istream &is, bool binary) {
 }
 
 void ContextDependency::GetPdfInfo(const std::vector<int32> &phones,
+                                   const std::vector<std::vector<std::pair<int32, int32> > > &pdf_class_pairs,
+                                   std::vector<std::vector<std::vector<std::pair<int32, int32> > > > *pdf_info) const {
+  // Todo
+  KALDI_ASSERT(pdf_info != NULL);
+  pdf_info->resize(1 + *std::max_element(phones.begin(), phones.end()));
+  EventType vec;
+  for (size_t i = 0 ; i < phones.size(); i++) {
+    // loop over phones
+    int32 phone = phones[i];
+    (*pdf_info)[phone].resize(pdf_class_pairs[phone].size());
+    for (size_t j = 0; j < pdf_class_pairs[phone].size(); j++) {
+      // loop over pdf_class pairs
+      vec.resize(2);
+      int32 pdf_class = pdf_class_pairs[phone][j].first,
+        self_loop_pdf_class = pdf_class_pairs[phone][j].second;
+      vec[0] = std::make_pair(static_cast<EventKeyType>(P_),
+                              static_cast<EventValueType>(phone));
+      vec[1] = std::make_pair(kPdfClass, static_cast<EventValueType>(pdf_class));
+      std::sort(vec.begin(), vec.end());
+      std::vector<EventAnswerType> forward_pdfs;  // forward pdfs that can be at this pos as this phone.
+      to_pdf_->MultiMap(vec, &forward_pdfs);
+      SortAndUniq(&forward_pdfs);
+
+      vec[0] = std::make_pair(static_cast<EventKeyType>(P_),
+                              static_cast<EventValueType>(phone));
+      vec[1] = std::make_pair(kPdfClass, static_cast<EventValueType>(self_loop_pdf_class));
+      std::sort(vec.begin(), vec.end());
+      std::vector<EventAnswerType> self_loop_pdfs;  // self-loop pdfs that can be at this pos as this phone.
+      to_pdf_->MultiMap(vec, &self_loop_pdfs);
+      SortAndUniq(&self_loop_pdfs);
+
+      // brute force enumeration
```

Comment:
Let's not use the term 'brute force' for this.. what I meant by brute force was, finding the exact list of pairs by enumerating all triphones (or n-phones, in general).  Let's call this 'overgeneration' of pairs.